scipy/sparse/_coo.py

```diff
@@ -286,13 +291,25 @@ def toarray(self, order=None, out=None):
         fortran = int(B.flags.f_contiguous)
         if not fortran and not B.flags.c_contiguous:
             raise ValueError("Output array must be C or F contiguous")
-        if self.ndim > 2:
-            raise ValueError("Cannot densify higher-rank sparse array")
         # This handles both 0D and 1D cases correctly regardless of the
         # original shape.
-        M, N = self._shape_as_2d
-        coo_todense(M, N, self.nnz, self.row, self.col, self.data,
-                    B.ravel('A'), fortran)
+        if self.ndim == 1:
+            coo_todense_nd(np.array([1]), self.nnz, self.ndim,
+                           self.coords[0], self.data, B.ravel('A'), fortran)
+        elif self.ndim == 2:
+            M, N = self._shape_as_2d
+            coo_todense(M, N, self.nnz, self.row, self.col, self.data,
+                        B.ravel('A'), fortran)
+            
+        else:
+            shape = np.array(self.shape)
+            if fortran:
+                strides = np.append(1, np.cumprod(shape[:-1]))
+            else:
+                strides = np.append(np.cumprod(shape[1:][::-1])[::-1], 1)
```

Comment:
A very minor tweak:  remove blank line and we can use the `self.shape` tuple to make `stride` to avoid a conversion.
```suggestion
        else:
            if fortran:
                strides = np.append(1, np.cumprod(self.shape[:-1]))
            else:
                strides = np.append(np.cumprod(self.shape[1:][::-1])[::-1], 1)
```